lib/pbench/server/auth/__init__.py

```diff
@@ -347,78 +507,5 @@ def get_userinfo(self, token: str = None) -> JSON:
                 "name": <full_name>
             }
         """
-
-        if token:
-            self.add_header_param("Authorization", f"Bearer {token}")
-
-        return self._get(self.USERINFO_ENDPOINT).json()
-
-    def _method(
-        self, method: str, path: str, data: Dict, **kwargs
-    ) -> requests.Response:
-        """Common frontend for the HTTP operations on OIDC client.
-
-        Args:
-            method : The API HTTP method
-            path : Path for the request.
-            data : Json data to send with the request in case of the POST
-            kwargs : Additional keyword args
-
-        Returns:
-            Response from the request.
-        """
-
-        try:
-            response = self.connection.request(
-                method,
-                urljoin(self.server_url, path),
-                params=kwargs,
-                data=data,
-                headers=self.headers,
-                verify=self.verify,
-            )
-            response.raise_for_status()
-            return response
-        except requests.exceptions.HTTPError:
-            raise OpenIDCAuthenticationError(response.status_code)
-        except (
-            requests.exceptions.ConnectionError,
-            requests.exceptions.Timeout,
-        ) as exc:
-            raise OpenIDClientError(
-                HTTPStatus.BAD_GATEWAY,
-                f"Failure to connect to the OpenID client, {self!r}: {exc}",
-            )
-        except Exception as exc:
-            raise OpenIDClientError(
-                HTTPStatus.INTERNAL_SERVER_ERROR,
-                f"Failure to complete the {method} request from the OpenID"
-                f" client, {self!r}: {exc}",
-            )
-
-    def _get(self, path: str, **kwargs) -> requests.Response:
-        """GET wrapper to handle an authenticated GET operation on the Resource
-        at a given path.
-
-        Args:
-            path : Path for the request.
-            kwargs : Additional keyword args
-
-        Returns:
-            Response from the request.
-        """
-        return self._method("GET", path, None, **kwargs)
-
-    def _post(self, path: str, data: Dict, **kwargs) -> requests.Response:
-        """POST wrapper to handle an authenticated POST operation on the
-        Resource at a given path.
-
-        Args:
-            path : Path for the request.
-            data : JSON request body
-            kwargs : Additional keyword args
-
-        Returns:
-            Response from the request.
-        """
-        return self._method("POST", path, data, **kwargs)
+        headers = {"Authorization": f"Bearer {token}"} if token else None
+        return self._connection.get(self._userinfo_endpoint, headers=headers).json()
```

Comment:
Yes, if the token is `None then this one is going to return `UNAUTHORIZED`. However, I don't think Peter made this change to the function, and I think updating the guts of this function should not be a part of this PR. 

I added this functionality long back when I was doing some experiments with OIDC APIs but I don't think we ever going to need to call `userinfo` endpoint as we pretty much get all the required information about the user when we decode the token so there is no need to call this endpoint at all. I will work on it in a separate PR.

Reply:
Given the history of this PR, I really cannot say who made the change, but, looking at the diffs here, someone replaced an `if` with a ternary...which I would normally support (😉) except that, in this particular case, it seems like immediately returning an error would be better than issuing the request (since we know that that is what will/should be the ultimate result).

Whether we should do that in this PR or not is the next question....